vsintegration/src/FSharp.Editor/LanguageService/SymbolHelpers.fs

```diff
@@ -67,69 +66,64 @@ module internal SymbolHelpers =
             return symbolUses
         }
 
-    let getSymbolUsesInProjects
-        (
-            symbol: FSharpSymbol,
-            projects: Project list,
-            onFound: Document -> range -> Async<unit>,
-            ct: CancellationToken
-        ) =
+    let getSymbolUsesInProjects (symbol: FSharpSymbol, projects: Project list, onFound: Document -> range -> CancellableTask<unit>) =
         match projects with
-        | [] -> Task.CompletedTask
+        | [] -> CancellableTask.singleton ()
         | firstProject :: _ ->
             let isFastFindReferencesEnabled = firstProject.IsFastFindReferencesEnabled
 
             // TODO: this needs to use already boxed boolean instead of boxing it every time.
             let props =
                 [| nameof isFastFindReferencesEnabled, isFastFindReferencesEnabled :> obj |]
 
-            backgroundTask {
+            cancellableTask {
                 // TODO: this needs to be a single event with a duration
                 TelemetryReporter.ReportSingleEvent(TelemetryEvents.GetSymbolUsesInProjectsStarted, props)
 
-                let tasks =
-                    [|
-                        for project in projects do
-                            yield
-                                project.FindFSharpReferencesAsync(symbol, onFound, "getSymbolUsesInProjects")
-                                |> CancellableTask.startAsTask ct
-                    |]
-
-                do! Task.WhenAll tasks
+                do!
+                    projects
+                    |> Seq.map (fun project -> project.FindFSharpReferencesAsync(symbol, onFound, "getSymbolUsesInProjects"))
+                    |> CancellableTask.whenAll
```

Comment:
I guess not a huge deal, but those were in an array for a reason - to avoid allocations (in this case it will be an enumerator, when mapping).
We want to allocate as little as possible in VS.